server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -522,15 +525,15 @@ public String executor() {
                 new IndicesModuleRequest(indexModule),
                 indicesModuleResponseHandler
             );
-            // TODO: make asynchronous
-            inProgressFuture.get(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+            inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
             logger.info("Received response from Extension");
-        } catch (Exception e) {
-            try {
-                throw e;
-            } catch (Exception e1) {
-                logger.error(e.toString());
+        } catch (CompletionException e) {
+            if (e.getCause() instanceof TimeoutException) {
+                logger.info("No response from extension to request.");
             }
+            throw e;
```

Comment:
The rethrowing here is not very useful, because the real cause is wrapped by the `CompletionException` I think we should rethrow `e.getCause()` all the time.

Reply:
e.getCause() is technically a Throwable, so I would have either have to add a throws declaration to the initalize and initializeExtension methods, or add a block like this:
```
try {
       throw e.getCause();
} catch (Throwable ex) {
       throw ex;
}
```
Which do you think is better?